### PR TITLE
use other diagnostic if available to minimize diff

### LINKED ISSUE #10798: Diff for VerifyDiagnostics is wrong
As an example:

``` diff
++>                 Diagnostic(ErrorCode.ERR_FixedBufferNotFixed, "MessageType").WithLocation(14, 19),
++>                 Diagnostic(ErrorCode.ERR_AnonDelegateCantUse, "x").WithArguments("x").WithLocation(20, 56),
++>                 Diagnostic(ErrorCode.ERR_BadArgType, "myStruct.MessageType").WithArguments("1", "char*", "char[]").WithLocation(26, 23)
-->                 Diagnostic(ErrorCode.ERR_FixedBufferNotFixed, "MessageType"),
-->                 Diagnostic(ErrorCode.ERR_AnonDelegateCantUse, "x").WithArguments("x")
```

Should be:

``` diff
++>                 Diagnostic(ErrorCode.ERR_BadArgType, "myStruct.MessageType").WithArguments("1", "char*", "char[]").WithLocation(26, 23)
```

### PR DESCRIPTION
**Pull request by TyOverby**
closes #10798

This is a very quick modification to the VerifyDiagnostics diff printer.  The change makes it so that it will try to produce a DiagnosticDescription in the "actual" output if an equivalent DiagnosticDescription occurs in "expected".

### CODE REVIEW

**Comment by TyOverby** [comment]
In my "intentionally sabotage a test" commit, you can see the results of this change.

Diff
```diff
 Diagnostic(ErrorCode.WRN_UnreferencedLabel, "a"),
 Diagnostic(ErrorCode.WRN_UnreferencedLabel, "b"),
++> Diagnostic(ErrorCode.WRN_UnreferencedLabel, "c").WithLocation(11, 7)
--> Diagnostic(ErrorCode.WRN_UnreferencedLabel, "d")
```

Under the old method, there would have been a subtraction on *every* line and re-adding of all those lines with a "WithLocation" call.  As you can see, the diff now only has the lines that are absolutely necessary.

**Comment by TyOverby** [comment]
@dotnet/roslyn-compiler 

Please review at your leisure.

File: src/Test/Utilities/Shared/Diagnostics/DiagnosticDescription.cs [hunk 1]
```diff
@@ -391,7 +389,7 @@ public static string GetAssertText(DiagnosticDescription[] expected, IEnumerable
 
             // Write out the 'command line compiler output' including squiggles (easy to read debugging info in the case of a failure).
             // This will be useful for VB, because we can't do the inline comments.
-            if (includeCompilerOutput)
+#if false
```
Comment (jaredpar) [inline src/Test/Utilities/Shared/Diagnostics/DiagnosticDescription.cs]: Please delete

**Comment by TyOverby** [comment]
@dotnet-bot test windows_debug_unit64_prtest  please

File: src/Test/Utilities/Shared/Diagnostics/DiagnosticDescription.cs (hunk 1 above)
Comment (TyOverby) [inline src/Test/Utilities/Shared/Diagnostics/DiagnosticDescription.cs]: Do you just want me to remove the entire thing that was ifdeffed?

Comment (jaredpar) [inline src/Test/Utilities/Shared/Diagnostics/DiagnosticDescription.cs]: If that's the right thing then yes.  We shouldn't be checking in `#if false` code.

File: src/Test/Utilities/Shared/Diagnostics/DiagnosticDescription.cs [hunk 2]
```diff
@@ -463,6 +439,11 @@ public static string GetAssertText(DiagnosticDescription[] expected, IEnumerable
                 }
 
                 var description = new DiagnosticDescription(d, errorCodeOnly: false, showPosition: true);
+                var idx = Array.IndexOf(expected, description);
+                if (idx != -1)
+                {
+                    description = expected[idx];
```
Comment (AlekseyTs) [inline src/Test/Utilities/Shared/Diagnostics/DiagnosticDescription.cs]: >description = expected[idx]; [](start = 20, length = 28)

Based on off-line conversation, this is going to affect content of the "Actual" portion of the output as well. If baseline doesn't have location for a particular diagnostics, the "Actual" won't show location any more. I think we would prefer to always have locations in the "Actual" portion. Ideally, all baselines should have locations and having them in "Actual" would make it easier to get to that state for existing tests over time.

**Comment by TyOverby** [comment]
@AlekseyTs 

```
Expected:
 Diagnostic(ErrorCode.WRN_UnreferencedLabel, "a"),
 Diagnostic(ErrorCode.WRN_UnreferencedLabel, "b"),
 Diagnostic(ErrorCode.WRN_UnreferencedLabel, "d")
Actual:
 // (11,1): warning CS0164: This label has not been referenced
 // a: b: c:Test2(Test1(out int x1), x1);
 Diagnostic(ErrorCode.WRN_UnreferencedLabel, "a").WithLocation(11, 1),
 // (11,4): warning CS0164: This label has not been referenced
 // a: b: c:Test2(Test1(out int x1), x1);
 Diagnostic(ErrorCode.WRN_UnreferencedLabel, "b").WithLocation(11, 4),
 // (11,7): warning CS0164: This label has not been referenced
 // a: b: c:Test2(Test1(out int x1), x1);
 Diagnostic(ErrorCode.WRN_UnreferencedLabel, "c").WithLocation(11, 7)
Diff:
 Diagnostic(ErrorCode.WRN_UnreferencedLabel, "a"),
 Diagnostic(ErrorCode.WRN_UnreferencedLabel, "b"),
++> Diagnostic(ErrorCode.WRN_UnreferencedLabel, "c").WithLocation(11, 7)
--> Diagnostic(ErrorCode.WRN_UnreferencedLabel, "d")
```

**Comment by cston** [comment]
LGTM

**Comment by AlekseyTs** [comment]
LGTM

**Comment by jcouv** [comment]
Relates to issue https://github.com/dotnet/roslyn/issues/10798

**Comment by TyOverby** [comment]
Merging into post-dev15

**Pull request merged**